src/experiments.js

```diff
@@ -47,6 +51,102 @@ export function isCanary(win) {
 }
 
 
+function getExperimentsKey_() {
+  let key;
+  const url = '' //TODO(kmh287): fill this in
+  if (url) {
+    return xhrFor(this.win).fetchJson(url, {
+      mode: 'cors',
+      method: 'GET',
+      ampCors: false,
+      credentials: 'omit',
+    }).then(jwkObj => {
+      if (isObject(jwkObj)) {
+        return this.crypto_.importPublicKey('experiments', key);
+      } else {
+        throw new Error(`Invalid response from experiments signing server`);
+      }
+    });
+  } else {
+    // The experiment signing service url wasn't found
+    return Promise.reject(
+        new Error(`Signing service for experiments does not exist.`));
+  }
+}
+
+/**
+ * Whether the specified experiment is on or off for origin trials. You should
+ * check if the experiment is already enabled before calling this function.
+ * @param {!Window} win
+ * @param {string} experimentId
+ * @return {!Promise<boolean>}
+ */
+export function isExperimentOnForOriginTrial(win, experimentId) {
```

Comment:
Should be up to the caller. May need to be run multiple times in the case where there are multiple AMP documents (e.g. shadow AMPs) but we should cache the result.